docs/samples/Microsoft.ML.AutoML.Samples/RecommendationExperiment.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.AutoML.Samples.DataStructures;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.AutoML.Samples
+{
+    public static class RecommendationExperiment
+    {
+        private static string TrainDataPath = "<Path to your train dataset goes here>";
+        private static string TestDataPath = "<Path to your test dataset goes here>";
+        private static string ModelPath = @"<Desired model output directory goes here>\Model.zip";
+        private static string LabelColumnName = "Rating";
+        private static uint ExperimentTime = 60;
+
+        public static void Run()
+        {
+            MLContext mlContext = new MLContext();
+
+            // STEP 1: Load data
+            IDataView trainDataView = mlContext.Data.LoadFromTextFile<Movie>(TrainDataPath, hasHeader: true, separatorChar: ',');
+            IDataView testDataView = mlContext.Data.LoadFromTextFile<Movie>(TestDataPath, hasHeader: true, separatorChar: ',');
+
+            var settings = new RecommendationExperimentSettings(RecommendationExperimentScenario.MatrixFactorization, "UserId", "MovieId")
+            {
+                MaxExperimentTimeInSeconds = ExperimentTime
+            };
+            var inputColumnInformation = new ColumnInformation();
+            inputColumnInformation.LabelCategoricalColumnNames.Add("MovieId");
+            inputColumnInformation.LabelCategoricalColumnNames.Add("UserId");
+            inputColumnInformation.LabelColumnName = "Rating";
+
+            // STEP 2: Run AutoML experiment
+            Console.WriteLine($"Running AutoML recommendation experiment for {ExperimentTime} seconds...");
+            ExperimentResult<RegressionMetrics> experimentResult = mlContext.Auto()
+                .CreateRecommendationExperiment(settings)
+                .Execute(trainDataView, inputColumnInformation);
+
+            // STEP 3: Print metric from best model
+            RunDetail<RegressionMetrics> bestRun = experimentResult.BestRun;
+            Console.WriteLine($"Total models produced: {experimentResult.RunDetails.Count()}");
+            Console.WriteLine($"Best model's trainer: {bestRun.TrainerName}");
+            Console.WriteLine($"Metrics of best model from validation data --");
+            PrintMetrics(bestRun.ValidationMetrics);
+
+            // STEP 5: Evaluate test data
+            IDataView testDataViewWithBestScore = bestRun.Model.Transform(testDataView);
+            RegressionMetrics testMetrics = mlContext.Regression.Evaluate(testDataViewWithBestScore, labelColumnName: LabelColumnName);
```

Comment:
same question here, MLContext does not have a `RecommendationCatalog Recommendation` property. Perhaps I should add this as well.

cc: @eerhardt @justinormont

Reply:
It does - but `Recommendation()` is an extension method on `MLContext`, and not a direct property.

https://github.com/dotnet/machinelearning/blob/3f98485e887177f2082f027c22f46fd64ebbe246/src/Microsoft.ML.Recommender/RecommenderCatalog.cs#L16-L19

https://github.com/dotnet/machinelearning/blob/3f98485e887177f2082f027c22f46fd64ebbe246/src/Microsoft.ML.Recommender/RecommenderCatalog.cs#L101-L117

Reply:
I found out why MLContext doesn't have RecommendationCatalog. Because there is this static helper used instead:
https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.Recommender/RecommenderCatalog.cs#L19

It's been used throughout the repo. So I'll keep it as is.

Reply:
Thanks @eerhardt just saw your reply. I will use Recommendation() instead.